### ISSUE TITLE
React Tutorial - Part 13 - Shopping Cart By React & Redux | Add Redux To Cart

### ISSUE THREAD

**Issue by RITESHKUMAR10** [basir/react-shopping-cart#17]
Hii Sir,
      I am facing an error like getState is not  a function in removeFromCart but in addToCart is working perfectly.
Please resolve this error .
     
                                                                                                                                                      Tanking you...
![Screenshot (252)](https://user-images.githubusercontent.com/43420487/90305884-14704300-dee5-11ea-97bf-3361738cb786.png)
![Screenshot (254)](https://user-images.githubusercontent.com/43420487/90305893-26ea7c80-dee5-11ea-888f-0c07edb66d0a.png)

**Comment by amabirbd** [comment]
I am facing the same problem. @basir  sir can you take a look at that issue?

**Comment by amabirbd** [comment]
ok, i solved the problem.

**Comment by yaprakbulut** [comment]
I have the same problem. How did you solve this ? @amabirbd

**Comment by amabirbd** [comment]
@yaprakbulut  most probably it's a typo  . I forgot a "_" .
I changed 
`x._id !== product.id` 
to 
`x._id !== product._id`

**Comment by yaprakbulut** [comment]
@amabirbd  No, this didn't solve. I checked the code but there is no wrong there. Also when my code keeps failing, I copied and pasted @basir ' s code, it still gives the same error.

**Comment by amabirbd** [comment]
> @amabirbd No, this didn't solve. I checked the code but there is no wrong there. Also when my code keeps failing, I copied and pasted @basir ' s code, it still gives the same error.

yes that happened to me too . check other files connected to that file like your cartReducer and store .

**Comment by amabirbd** [comment]
@yaprakbulut  and most importantly, don't forget to restart your pc  ;)

**Comment by ebbie** [comment]
I got the same error too. Not sure how to solve it. @yaprakbulut Did you find the solution ?
@amabirbd I checked my code and there seems to be no typo

**Comment by ebbie** [comment]
@RITESHKUMAR10 @amabirbd @yaprakbulut
I think I found my solution too. 
In /components/Cart.js,

I missed adding {} around removeFromCart

Previously it was,
export default connect((state) => ({
    cartItems: state.cart.cartItems,
}),
    removeFromCart
)(Cart); 

and I changed the above code to 

export default connect((state) => ({
    cartItems: state.cart.cartItems,
}),
    {removeFromCart} // This is where I forgot to add the {}
)(Cart);

@amabirbd I agree with you. Looks like only if we are missing something in the connected files, we are getting this error.

PS: If you are following udemy course videos and doing exactly as the professor says then it looks like the instructor forgot to add these curly braces {} while coding. 

Hope this helps

**Comment by amabirbd** [comment]
@ebbie that's probably it.
 Are you taking the udemy course?

**Comment by yaprakbulut** [comment]
> @RITESHKUMAR10 @amabirbd @yaprakbulut
> I think I found my solution too.
> In /components/Cart.js,
> 
> I missed adding {} around removeFromCart
> 
> Previously it was,
> export default connect((state) => ({
> cartItems: state.cart.cartItems,
> }),
> removeFromCart
> )(Cart);
> 
> and I changed the above code to
> 
> export default connect((state) => ({
> cartItems: state.cart.cartItems,
> }),
> {removeFromCart} // This is where I forgot to add the {}
> )(Cart);
> 
> @amabirbd I agree with you. Looks like only if we are missing something in the connected files, we are getting this error.
> 
> PS: If you are following udemy course videos and doing exactly as the professor says then it looks like the instructor forgot to add these curly braces {} while coding.
> 
> Hope this helps

@ebbie 
Thank you so much! It worked :)

**Comment by RITESHKUMAR10** [comment]
Thank you so much,its working.

**Comment by ebbie** [comment]
> @ebbie that's probably it.
> Are you taking the udemy course?

@amabirbd Yes, I completed it yesterday and the course was really good.
